pkg/scheduler/dependency_graph.go

```diff
@@ -217,28 +236,32 @@ func (sched Scheduler) newScheduledResource(kind, name string, resDefs map[strin
 		return nil, fmt.Errorf("not a proper resource kind: %s. Expected '%s'",
 			kind, strings.Join(resources.Kinds, "', '"))
 	}
-	r, err := sched.newResource(kind, name, resDefs, gc, resourceTemplate)
+	r, existing, err := sched.newResource(kind, name, resDefs, gc, resourceTemplate, silent)
 	if err != nil {
 		return nil, err
 	}
 
-	return newScheduledResourceFor(r, gc), nil
+	return newScheduledResourceFor(r, gc, existing), nil
 }
 
 func (sched Scheduler) newResource(kind, name string, resDefs map[string]client.ResourceDefinition,
-	gc *GraphContext, resourceTemplate interfaces.ResourceTemplate) (interfaces.Resource, error) {
+	gc *GraphContext, resourceTemplate interfaces.ResourceTemplate, silent bool) (interfaces.Resource, bool, error) {
```

Comment:
I'm going to create `GetDependencyGraph`  helper method that calls `BuildDependencyGraph` in no-side-effects mode to make it more explicit